lib/widgets/buttons.dart

```diff
@@ -265,11 +266,12 @@ class _AppButtonState extends State<AppButton> with SingleTickerProviderStateMix
             children: [
               widget.icon!,
               const SizedBox(width: 6.0),
-               Padding(
-                padding: const EdgeInsets.only(bottom: 1.0),
-                child: _buildText(),
+              Flexible(
+                child: Padding(
+                  padding: const EdgeInsets.only(bottom: 1.0),
+                  child: _buildText(),
+                ),
               ),
-              const SizedBox(width: 8.0),
```

Comment:
| With offset | Without offset |
| - | - |
| ![image](https://user-images.githubusercontent.com/39104740/168331097-3468d87d-df3a-4268-b801-2a06b7f6f0da.png) | ![image](https://user-images.githubusercontent.com/39104740/168331126-258fde53-cb5a-4300-bb76-28d4258a4a63.png) |

> Therefore, the [icon, SizedBox, Flexible] group is now perfectly centered. The additional SizedBox at the end acted as padding with caused the icon and text to be off-centered.

It's clearly not perfectly centered, because of the padding that icons add, as you also mentioned below in your comment. Hence this padding should be kept there, it is there to center it visually.

If there's not enough space because of padding, we should just reduce the overall `ElevatedButton` (/`TextButton`) padding when there's an icon, and/or decrease the font size a bit when the text is too long, and removing this particular padding would not solve the problem of "Zufallswiedergabe" not fitting into this button on most screens anyway.

There's also such thing as AutoSizeText which probably could help to do this automatically.

Reply:
> It's clearly not perfectly centered,

I agree that we need padding, but I'm suggesting that the padding shouldn't be constant. It's not enough to just look at the play button, as it is an extreme edge case and the padding affects all `AppButton`s:

| Description | Screenshot |
|--------------|-------------|
| Without offset | ![AppButton - Before](https://user-images.githubusercontent.com/6966049/168492846-6f3a8d83-36c5-45c3-967f-eeb8da78acb5.png) |
| With offset | ![AppButton - After](https://user-images.githubusercontent.com/6966049/168492850-57213a02-9276-48fb-bf8c-f168c9d64bff.png) |
| Without offset (Padding in pixels) | ![AppButton - After - Measurements](https://user-images.githubusercontent.com/6966049/168492852-be4165a5-13e4-49bd-b55e-352e8fdafb54.png) |
| With offset (Padding in pixels) | ![AppButton - Before - Measurements](https://user-images.githubusercontent.com/6966049/168492855-69a11e2a-55cf-462a-aff2-4aa90480d1de.png) |

As you can see with the measurements, centering the play icon off-centers the shuffle icon. The different icons require different padding, and we should account for that.

> AutoSizeText

Yes, that might be the final solution to the overflow problem. I will look into it.

The reason I made this change is because the when the shuffle button is filled with text, it became very noticeable that it wasn't centered and it felt like there was some wasted space at the end. I realize that this doesn't have anything to do with the original intention of the PR anymore, so I will revert this and make a dedicated PR for it.

Reply:
The text in the `AppButton` is now automatically scaled:
![Artist view](https://user-images.githubusercontent.com/6966049/168494095-271d8c5c-d933-4b2b-b05c-f2bb1c086316.png)